src/github.ts

```diff
@@ -281,6 +284,84 @@ export class GitHub {
     }
   }
 
+  async commitsSinceShaRest(
+    sha?: string,
+    path?: string,
+    per_page = 100
+  ): Promise<Commit[]> {
+    let page = 1;
+    let found = false;
+    const baseBranch = await this.getDefaultBranch();
+    const commits = [];
+    while (!found) {
+      const response = await this.request(
+        'GET /repos/{owner}/{repo}/commits{?sha,page,per_page,path}',
+        {
+          owner: this.owner,
+          repo: this.repo,
+          sha: baseBranch,
+          page,
+          per_page,
+          path,
+        }
+      );
+      for (const commit of (response as CommitsListResponse).data) {
+        if (commit.sha === sha) {
+          found = true;
+          break;
+        }
+        // skip merge commits
+        if (commit.parents.length === 2) {
+          continue;
+        }
+        commits.push([commit.sha, commit.commit.message]);
+      }
+      page++;
+    }
+    const ret = [];
+    for (const [ref, message] of commits) {
+      const files = [];
+      let page = 1;
+      let moreFiles = true;
+      while (moreFiles) {
+        // the "Get Commit" resource is a bit of an outlier in terms of GitHub's
+        // normal pagination: https://git.io/JmVZq
+        // The behavior is to return an object representing the commit, a
+        // property of which is an array of files. GitHub will return as many
+        // associated files as there are, up to a limit of 300, on the initial
+        // request. If there are more associated files, it will send "Links"
+        // headers to get the next set. There is a total limit of 3000
+        // files returned per commit.
+        // In practice, the links headers are just the same resourceID plus a
+        // "page=N" query parameter with "page=1" being the initial set.
+        //
+        // TODO: it is more robust to follow the link.next headers (in case
+        // GitHub ever changes the pattern) OR use ocktokit pagination for this
+        // endpoint when https://git.io/JmVll is addressed.
```

Comment:
I know this is kind of hacky/rookie... but do you mind if I just add a tracking issue to harden/future-proof it (maybe add to #830)?